langchain/schema.py

```diff
@@ -261,6 +266,10 @@ def add_user_message(self, message: str) -> None:
     def add_ai_message(self, message: str) -> None:
         """Add an AI message to the store"""
 
+    @abstractmethod
+    def add_message(self, message: BaseMessage) -> None:
```

Comment:
This seems reasonable to me. Looks more appropriate than add_ai_message or add_human_message for the lowest building block of chat history.

Reply:
went a step further now, got rid of all implementations of add_xx_message, instead using add_message in each subclass, and then noticed most had another method (usually append) with the same footprint, so got rid of those as well, much cleaner IMHO!

Reply:
i dont think this is super backwards compatible, can we `raise NotImplementedError` rather than have as abstract? that way if people subclassed it it doesnt break